add a collector for Eventstore Projections

merlijntishauser opened the pull request:
In this PR I have added a collector for [Eventstore](https://geteventstore.com/) projections. It has only be tested against a single node version of eventstore (not clustered). We use it already in production and does it's job nicely in combination with hostedGraphite. Note that $ in projection names are replaced by a _ to avoid problems in Grafana.

The collector is based on the HTTPJSON collector, which could not handle the slightly more complex json output from the eventstore api.

jaingaurav commented on src/collectors/eventstoreprojections/eventstoreprojections.py:
```diff
@@ -0,0 +1,100 @@
+        )
+        default_config.update({
+            'path': 'eventstore',
+            'url': 'http://hostname:2113/projections/all-non-transient',
```
could you change this from `hostname` to `localhost`? At least then it is somewhat of a legitimate default config.

jaingaurav commented on src/collectors/eventstoreprojections/eventstoreprojections.py:
```diff
@@ -0,0 +1,100 @@
+        config_help.update({
+            'url': 'Full URL',
+            'headers': 'Header variable if needed. '
+            'Will be added to every request',
```
Could you please keep the lines of the description at the same indent level? This might require wrapping it in parenthesis.

jaingaurav commented on src/collectors/eventstoreprojections/eventstoreprojections.py:
```diff
@@ -0,0 +1,100 @@
+                    yield ("%s.%s" % (prefix, key), value)
+                else:
+                    # very chatty, uncomment if you want to see the output
+                    # self.log.debug("ignoring string value = %s", value)
```
Rather than keeping this uncommented, please add a `debug` config option

jaingaurav commented:
Gave some very minor comments mainly around the config options. Please make the edits and I'll merge this in.

merlijntishauser commented:
Added a commit where the comments are addressed. Thanks for the feedback!

jaingaurav commented on src/collectors/eventstoreprojections/eventstoreprojections.py:
```diff
@@ -0,0 +1,110 @@
+        )
+        default_config.update({
+            'path': "eventstore",
+            'url': 'http://localhost',
```
So I still feel like this option should be `hostname` instead of url. Might be best to break it up into 2 options `protocol` and `hostname`

jaingaurav commented on src/collectors/eventstoreprojections/eventstoreprojections.py:
```diff
@@ -0,0 +1,110 @@
+
+                data = {}
+                for projection in projections:
+                    name = projection["name"].replace("$", "_")
```
Does it make sense to replace `$` with `_` or simply just delete the `$`. The delete option would make nicer metric names, however I don't know if there could be a name collision.

merlijntishauser commented:
Split up the url into 2 new config options as proposed, I misunderstood the initial comment. 
Just deleting the $ would almost for sure give name collisions. As a solution, I introduced a config option so that you, if you really want, could skip replacing the $ or replace it by a string of your choice.
As done in the rabbitMQ collector for dots and/or slashes.

Unfortunately, travis fails on the updated PR for python 2.6 with the message: "The command "pip install -r .travis.requirements.txt" failed and exited with 1 during ." It seems very unlikely this is caused by my changes in the last 2 commits. So it should be safe to merge ;)

jaingaurav commented:
Thanks a lot, I re-ran travis tests and it looks good now. The failure you mentioned is a known flakiness.
